prov/util/src/util_mr_monitor.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2017 Cray Inc. All rights reserved.
+ * Copyright (c) 2017 Intel Inc. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <ofi_mr.h>
+
+
+int ofi_mr_monitor_init(struct ofi_mr_monitor *mr_monitor)
+{
+	fastlock_init(&mr_monitor->lock);
+	dlist_init(&mr_monitor->eq_list);
+	return FI_SUCCESS;
+}
+
+void ofi_mr_monitor_finalize(struct ofi_mr_monitor *mr_monitor)
+{
+	assert(dlist_empty(&mr_monitor->eq_list));
+}
+
+void ofi_mr_monitor_eq_add(struct ofi_mr_monitor *mr_monitor,
+			   struct ofi_mr_monitor_eq *eq,
+			   int events)
+{
+	eq->mr_monitor = mr_monitor;
+	eq->events = events;
+	fastlock_init(&eq->lock);
+	dlist_init(&eq->list);
+
+	fastlock_acquire(&mr_monitor->lock);
+	dlist_insert_tail(&eq->entry, &mr_monitor->eq_list);
+	fastlock_release(&mr_monitor->lock);
+}
+
+void ofi_mr_monitor_eq_del(struct ofi_mr_monitor_eq *eq)
+{
+	assert(dlist_empty(&eq->list));
+
+	fastlock_acquire(&eq->mr_monitor->lock);
+	dlist_remove(&eq->entry);
+	fastlock_release(&eq->mr_monitor->lock);
+
+	fastlock_destroy(&eq->lock);
+}
+
+static void util_mr_monitor_read_events(struct ofi_mr_monitor_eq *eq)
```

Comment:
I think this was commented on in the last PR. Is this going to stay as as 'util_mr_monitor_*' instead of 'ofi_mr_monitor'?

Reply:
The Utility code uses the following style as I noticed:
- external functions (like API) are declared with `ofi_[utility_module]_*` prefix
- internal functions (that is used in this file only) are declared with `util_[utility_module]_*` prefix

For example, looking at the CQ utility code (util_cq.c):
`ssize_t ofi_cq_read(struct fid_cq *cq_fid, void *buf, size_t count)`
`static void util_cq_read_tagged(void **dst, void *src)`